### PR TITLE
Fix remove iv needing power to operate.

### PR DESCRIPTION
**Pull request by Huffie56**
# About the pull request
fixes: https://github.com/cmss13-devs/cmss13/issues/3848
and add a skill check like for the blood pack.
also turned four variable with a single letter into better ones.

# Explain why it's good for the game
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
add: Add a skill check to operate IV.
fix: Fix remove iv needing power to operate.
/:cl:

### CODE REVIEW

**Comment by Huffie56** [comment]
the var to make a check skill can't be use there so i am just going to remove the skill check.....

**Comment by Drulikar** [comment]
> the var to make a check skill can't be use there so i am just going to remove the skill check.....

That's because the variables have terrible names in this proc. You likely need usr not user.

**Comment by Huffie56** [comment]
> > the var to make a check skill can't be use there so i am just going to remove the skill check.....
> 
> That's because the variables have terrible names in this proc. You likely need usr not user.

image

it's not user causing issue it seem it's the skill check... will keep looking into it....

**Comment by Huffie56** [comment]
i have no idea on how to fix the check skill issue so i will focus on making IV work without power for now...

File: code/game/machinery/iv_drip.dm [hunk 1]
```diff
@@ -68,7 +72,11 @@
 		var/mob/living/carbon/human/H = usr
 		if(H.stat || get_dist(H, src) > 1 || H.blinded || H.lying)
 			return
-
+/*
+		if(!skillcheck(user, SKILL_SURGERY, SKILL_SURGERY_NOVICE))
+			to_chat(user, SPAN_WARNING("You don't know how to connect this!"))
+			return
+*/
```
Comment (realforest2001) [inline code/game/machinery/iv_drip.dm:79]:
```suggestion

		if(!skillcheck(user, SKILL_SURGERY, SKILL_SURGERY_NOVICE))
			to_chat(user, SPAN_WARNING("You don't know how to connect this!"))
			return

```

File: code/game/machinery/iv_drip.dm [hunk 2]
```diff
@@ -68,7 +72,11 @@
 		var/mob/living/carbon/human/H = usr
 		if(H.stat || get_dist(H, src) > 1 || H.blinded || H.lying)
```
Comment (realforest2001) [inline code/game/machinery/iv_drip.dm:73]:
```suggestion
		var/mob/living/carbon/human/user = usr
		if(user.stat || get_dist(user, src) > 1 || user.blinded || user.lying)
```

File: code/game/machinery/iv_drip.dm [hunk 3]
```diff
@@ -68,7 +72,11 @@
...
 			return
-
+/*
+		if(!skillcheck(user, SKILL_SURGERY, SKILL_SURGERY_NOVICE))
+			to_chat(user, SPAN_WARNING("You don't know how to connect this!"))
+			return
+*/
 		if(attached)
 			H.visible_message("[H] detaches \the [src] from \the [attached].", \
```
Comment (realforest2001) [inline code/game/machinery/iv_drip.dm:81]:
```suggestion
			user.visible_message("[user] detaches \the [src] from \the [attached].", \
```

File: code/game/machinery/iv_drip.dm [hunk 4]
```diff
@@ -68,7 +72,11 @@
...
 			"You detach \the [src] from \the [attached].")
```
Comment (realforest2001) [inline code/game/machinery/iv_drip.dm:82]: You'll need to go through and replace the other instances of "H" with "user"

**Review by realforest2001 — CHANGES_REQUESTED** [review]

**Comment by Drulikar** [comment]
> i have no idea on how to fix the check skill issue so i will focus on making IV work without power for now...

As I mentioned, and as you can see from Forest's suggestions, you can't merely copy and paste code and expect every variable to be named the same thing. In your image, each error is because `user` does not exist for the same thing it did from where you copied this code.

image

**Comment by Huffie56** [comment]
will take a look at this in 10h i didn't feel like sinking 1-2h into this today.

File: code/game/machinery/iv_drip.dm [hunk 5]
```diff
@@ -151,20 +145,20 @@
...
 
-			if(!istype(T))
+			if(!istype(patient))
 				return
-			if(ishuman(T))
-				var/mob/living/carbon/human/H = T
-				if(H.species && H.species.flags & NO_BLOOD)
+			if(ishuman(patient))
+				var/mob/living/carbon/human/user = patient
+				if(user.species && user.species.flags & NO_BLOOD)
```
Comment (realforest2001) [inline code/game/machinery/iv_drip.dm:154]:
```suggestion
				var/mob/living/carbon/human/human_patient = patient
				if(human_patient.species && human_patient.species.flags & NO_BLOOD)
```

**Review by realforest2001 — CHANGES_REQUESTED** [review]

**Review by realforest2001 — APPROVED** [review]

File: code/game/machinery/iv_drip.dm [hunk 6]
```diff
@@ -44,33 +46,25 @@
...
 
 	if(ishuman(usr))
-		var/mob/living/carbon/human/H = usr
-		if(H.stat || get_dist(H, src) > 1 || H.blinded || H.lying)
+		var/mob/living/carbon/human/user = usr
+		if(user.stat || get_dist(user, src) > 1 || user.blinded || user.lying)
+			return
+
+		if(!skillcheck(user, SKILL_SURGERY, SKILL_SURGERY_NOVICE))
+			to_chat(user, SPAN_WARNING("You don't know how to connect this!"))
```
Comment (Drulikar) [inline code/game/machinery/iv_drip.dm:63]:
```suggestion
			to_chat(user, SPAN_WARNING("You don't know how to [attached ? "disconnect" : "connect"] this!"))
```

**Pull request merged**